### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by aspidites**
# Description

This PR (once complete) adds tests for functionality, which should help guide creation of new backends. I determined which tests to write by following the tutorials
# Progress

I'll update this list soon to include all tests I plan to write to make it easier to track progress of this PR.
- [x] table creation
- [ ] row insertion

### CODE REVIEW

File: beam.cabal [hunk 1]
```diff
@@ -2,7 +2,7 @@
 -- see http://haskell.org/cabal/users-guide/
 
 name:                beam
-version:             0.4.0.0
+version:             0.4.1.0
```
Comment (aspidites) [inline beam.cabal]: I tried to follow PVP here. We didn't change the type signature of anything, and only added new functionality, so the minor version was bumped.

File: src/Database/Beam/Backend.hs [hunk 2]
```diff
@@ -116,11 +116,17 @@ openDatabase' isDebug db strat dbSettings =
...
 
+
 dumpSchema :: Database db => DatabaseSettings db -> IO ()
-dumpSchema (db :: DatabaseSettings db) =
+dumpSchema (db :: DatabaseSettings db) = do
+  putStrLn "Dumping database schema ..."
+  schema <- showSchema db
+  mapM_ putStrLn $ lines schema
+
+showSchema :: Database db => DatabaseSettings db -> IO String
```
Comment (aspidites) [inline src/Database/Beam/Backend.hs]: As dumpSchema returned IO (), I couldn't capture the result, which I needed for tests.

**Comment by aspidites** [comment]
@tathougies can you offer any guidance into if there is a simple way to capture the sql generated by a given command. I see that debug options allow those to go to console, but I'd prefer a pure function that just spit out potential SQL, if possible.

**Comment by tathougies** [comment]
@aspidites Sorry for the long delay! Had to take a break for personal reasons :)

In terms of getting the command for a particular expression.. There's currently no way to do that, but it would be straightforward to add. If you look in Database/Beam/Query.hs, you'll see all the functions we can use to query/update the database. Each of these (query, insertInto, updateWhere, etc) has an associated function (queryToSQL', insertToSQL, updateToSQL) respectively which returns a SQLSelect, SQLInsert, or SQLUpdate respectively. You can use these objects to construct a SQLCommand, which can be pretty printed using the Database.Beam.Query.ppSQL function.

Let me know which functions you think you'd need, and I can take a stab at them.

**Comment by aspidites** [comment]
Thanks. I'll give it ago myself and ping you if I have trouble. May be a
minute. Going through my own personal stuff now so remaining motivated has
been... difficult as of late.

**Pull request closed without merging**